driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/gridfs/GridFSUploadPublisherImpl.java

```diff
@@ -228,40 +205,37 @@ private <T> Mono<String> createIndexMono(final MongoCollection<T> collection, fi
     }
 
     private Mono<Long> createSaveChunksMono(final AtomicBoolean terminated, @Nullable final Timeout timeout) {
-        return Mono.create(sink -> {
```

Comment:
Remove the use of sink and use the project reactor api to declare how the data is processed.